pkg/didcomm/protocol/presentproof/service.go

```diff
@@ -484,25 +489,42 @@ func (s *Service) deleteTransitionalPayload(id string) error {
 
 // Actions returns actions for the async usage.
 func (s *Service) Actions() ([]Action, error) {
-	records := s.store.Iterator(
-		fmt.Sprintf(transitionalPayloadKey, ""),
-		fmt.Sprintf(transitionalPayloadKey, storage.EndKeySuffix),
-	)
-	defer records.Release()
+	records, err := s.store.Query(transitionalPayloadKey)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query store: %w", err)
+	}
+
+	defer func() {
```

Comment:
we could use a static helper function to make it a bit easier and less repetitive...

```go
defer storage.Close(records, logger)  // logger should be optional
```

Reply:
Where would the static helper function go? An `internal` file perhaps?

Reply:
Not `internal` - downstream packages won't be able to use it.

I would place it right in `storage`.

Reply:
Done - since spi/storage is in a different module, I had to do it in a separate commit... please see https://github.com/hyperledger/aries-framework-go/pull/2566

Reply:
Good idea - this should also help reduce the number of lines needed to be unit tested

Reply:
I've updated this PR to use that new helper method